### ISSUE TITLE
Insufficient punctuation rules for CJK

### ISSUE THREAD

**Issue by z-rui** [XmacsLabs/mogan#397]
Source:
```
<tabular|<tformat|<cwith|1|1|1|1|cell-width|1in>|<cwith|1|1|1|1|cell-hmode|exact>|<cwith|1|1|1|1|cell-hyphen|t>|<table|<row|<\cell>
  \<#6211\>\<#80FD\>\<#541E\>\<#4E0B\>\<#FF08\>\<#73BB\>\<#7483\>\<#FF09\>\<#800C\>\<#4E0D\>\<#4F24\>\<#8EAB\>\<#4F53\>\<#3002\>

  \<#6211\>\<#80FD\>\<#541E\>\<#4E0B\>\<#73BB\>\<#7483\>\<#FF08\>\<#800C\>\<#4E0D\>\<#FF09\>\<#4F24\>\<#8EAB\>\<#4F53\>\<#3002\>

  \<#6211\>\<#80FD\>\<#541E\>\<#4E0B\>\<#73BB\>\<#7483\>\<#201C\>\<#800C\>\<#4E0D\>\<#201D\>\<#4F24\>\<#8EAB\>\<#4F53\>\<#3002\>

  \<#6211\>\<#80FD\>\<#541E\>\<#4E0B\>\<#73BB\>\<#7483\>\<#800C\>\<#2014\>\<#2014\>\<#4E0D\>\<#4F24\>\<#8EAB\>\<#4F53\>\<#3002\>

  \<#6211\>\<#80FD\>\<#541E\>\<#4E0B\>\<#3010\>\<#73BB\>\<#7483\>\<#3011\>\<#800C\>\<#4E0D\>\<#4F24\>\<#8EAB\>\<#4F53\>\<#3002\>
</cell>>>>>
```

Output:
<img width="120" alt="image" src="https://user-images.githubusercontent.com/7368209/204891440-17842162-efdb-4d39-9d60-338d784663e2.png">

Problems:

- Punctuations like “, （ and 【 should not be allowed at the end of a line.
- Punctuations like ”, ） and 】 should not be allowed at the beginning of a line.

**Comment by darcy-shen** [comment]
The generated ID is 111 for this bug.

**Comment by darcy-shen** [comment]
Documented in #456

**Comment by darcy-shen** [comment]
Before #510 

<img width="699" alt="image" src="https://user-images.githubusercontent.com/1267865/213927267-c96dc871-2cf4-4300-bdc1-952767b38307.png">


After #510 

<img width="710" alt="image" src="https://user-images.githubusercontent.com/1267865/213927301-5cf5fca1-f29c-476c-a077-89eaaf8d57f3.png">


According to https://www.w3.org/International/clreq/#prohibition_rules_for_line_start_end and the screenshots, there only one symbol to be fixed:

<img width="97" alt="image" src="https://user-images.githubusercontent.com/1267865/213927387-6b4c217f-5b82-457f-8fcf-1bac6e3063ea.png">

**Comment by darcy-shen** [comment]
The remaining problem is about encoding, let me close the ticket now.

closed by #510
